File: reactjs/src/services/GoodIssues/dto/GoodIssuesDto.ts
import { PagedFilterAndSortedRequest } from "../../dto/pagedFilterAndSortedRequest"

export interface PagedResultGooodsIssuesRequestDto extends PagedFilterAndSortedRequest  {
  keyword: string
  FromPlant:string
}


export interface GetAllGoodIssues{
    id: number,
    transactionNumber: string,
    transactionType: string,
    materialNumber: string,
    transactionQuantiry:number,
    materialLot: string,
    materialType: string,
    fromPlant: string,
    fromSubLocation: string,
    toPlant: string,
    toSubLocation: string,
    docmentType: string,
    total: number,
    isCompleted: true,
    children: [
        {
          id: number,
          transtiomId: number,
          issuesType: string,
          transaction: string,
          issueQuantity: 0,
          materialNumber: string,
          materialType: string,
          materialLot: string,
          plant: string,
          subLocation: string,
          isOnhandsProcessed: true
        }
      ]
}


export interface GetGoodIssuesOutput{
    id: number,
    transactionNumber: string,
    transactionType: string,
    materialNumber: string,
    transactionQuantiry:number,
    materialLot: string,
    materialType: string,
    fromPlant: string,
    fromSubLocation: string,
    toPlant: string,
    toSubLocation: string,
    docmentType: string,
    total: number,
    isCompleted: true,
    children: [
        {
          id: number,
          transtiomId: number,
          issuesType: string,
          transaction: string,
          issueQuantity: 0,
          materialNumber: string,
          materialType: string,
          materialLot: string,
          plant: string,
          subLocation: string,
          isOnhandsProcessed: true
        }
      ]
}

export interface CreateOrUpdateGoodIssuesInput{
    id:number
    transactionID: number,
    isOnhandsProcessed: boolean
    issueQuantity: number
}

export interface UpdateGoodIssuesInput{
    id:number,
    issueQuantity: number,
    
}

File: aspnet-core/src/emes.Application/Services/Transaction/GoodIssues/Dto/GoodIssuesDto.cs


namespace emes.Services.Transaction.GoodIssues.Dto
{
    [AutoMapFrom(typeof(GoodIssuesModel))]
    public class GoodIssuesDto:EntityDto<Guid>
    {
      
        public Guid TranstiomId { get; set; }
        public string IssuesType { get; set; }
        public string Transaction { get; set; }
        public decimal IssueQuantity { get; set; }
        public string MaterialNumber { get; set; }
        public string MaterialType { get; set; }
        public string MaterialLot { get; set; }
        public string Plant { get; set; }
        public string SubLocation { get; set; }
        public bool IsOnhandsProcessed { get; set; }
    }
}
